smoke-tests/src/test/java/com/splunk/opentelemetry/logger/ResourceAttributesForLoggingTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.logger;
+
+import com.splunk.opentelemetry.SmokeTest;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Main feature why we expose resource attributes as system properties is to be able to use them in
+ * logs. This is a smoke test, because there are several moving parts involved. If everything
+ * together doesn't work, there is no point in any simpler unit/integration test.
+ */
+public class ResourceAttributesForLoggingTest {
+
+  @Test
+  void resourceAttributesCanBeLogged() throws IOException, InterruptedException {
+    Assumptions.assumeTrue(SmokeTest.agentPath != null, "Agent path is specified");
+
+    Process process = startProcessWithJavaagent();
+
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    transfer(process.getInputStream(), out);
+    ByteArrayOutputStream err = new ByteArrayOutputStream();
+    transfer(process.getErrorStream(), err);
+
+    Assertions.assertTrue(process.waitFor(30L, TimeUnit.SECONDS), "Process exited");
+
+    int exitValue = process.exitValue();
+    if (exitValue != 0) {
+      System.err.println(err.toString());
+    }
```

Comment:
Nitpicking: you can use `Assertions#assertEquals(int, int, Supplier<String>)` to lazily compute assertion message instead of an if statement here.

Reply:
This wasn't even about lazy evaluation. It was about having stderr in logs, if exit code was non-zero. But having it in the assertion message is even better, indeed!